src/SqsPoller/SqsPollerHostedService.cs

```diff
@@ -117,7 +126,7 @@ public async Task HandleMessage(Message message, CancellationToken cancellationT
             }
             catch (Exception e)
             {
-                _logger.Log(_config.ExceptionDefaultMessageLogLevel, e, "Failed to receive messages from the queue"); ;
+                _config.OnException?.Invoke(e, "Failed to receive messages from the queue");
```

Comment:
I don't think that one exception handler `config.OnException` is a good idea to different kind of exceptions(
delete  a message faulted, a message is not handled, messages are not read from a queue). I suggest implementing `OnException` per type or passing ExceptionType enum in parameters instead of message.

Reply:
@AlexeyBogdan95 Updated, please check & approve PR